src/message-list/MessageList.js

```diff
@@ -45,6 +81,27 @@ export default class MessageList extends React.PureComponent {
     registerAppActivity(auth);
   }
 
+  // Checks if the current scroll position is beyond threshold scroll position
+  // for showing new messages indicator
+  hasScrolledBeyondInfoThreshold() {
+    return this.scrollOffset > NEW_MSG_INFO_SCROLL_THRESHOLD;
+  }
+
+  // Checks if new messages have been received
+  didReceiveNewMessages(props) {
+    return props.messages.length > this.props.messages.length;
```

Comment:
This is not sufficient check, I will write some code to do the checks (we currently don't have any)